File: simulator/event.py
import itertools

class Event:
    _ids = itertools.count()

    def __init__(self, time, event_type, server=None, job_id=None, job_class=None):
        self.id = next(Event._ids)
        self.time = time
        self.event_type = event_type  # "ARRIVAL" o "DEPARTURE"
        self.server = server
        self.job_id = job_id
        self.job_class = job_class
        self.cancelled = False

    def __lt__(self, other):
        if self.time == other.time:
            return self.id < other.id
        return self.time < other.time

    def __repr__(self):
        return (f"Event(t={self.time:.4f}, type={self.event_type}, "
                f"server={self.server}, job={self.job_id}, class={self.job_class})")

File: java/NextEventSimulator/src/main/java/com/g45/webappsim/simulator/Event.java
package com.g45.webappsim.simulator;

import java.util.concurrent.atomic.AtomicLong;

public final class Event implements Comparable<Event> {


    public enum Type { ARRIVAL, DEPARTURE }

    private static final AtomicLong SEQ = new AtomicLong(0);

    private final long id;                 // increasing to break ties
    private double time;
    private final Type type;
    private final String server;           // node name
    private final int jobId;               // -1 means external arrival
    private final int jobClass;            // 1,2,3,...
    private volatile boolean cancelled;    // soft-cancel flag

    public Event(double time, Type type, String server, int jobId, int jobClass) {
        this.id = SEQ.getAndIncrement();
        this.time = time;
        this.type = type;
        this.server = server;
        this.jobId = jobId;
        this.jobClass = jobClass;
        this.cancelled = false;
    }

    public long getId() { return id; }
    public double getTime() { return time; }
    public void setTime(double t) { this.time = t; }
    public Type getType() { return type; }
    public String getServer() { return server; }
    public int getJobId() { return jobId; }
    public int getJobClass() { return jobClass; }
    public boolean isCancelled() { return cancelled; }
    public void cancel() { this.cancelled = true; }

    @Override public int compareTo(Event o) {
        int cmp = Double.compare(this.time, o.time);
        if (cmp != 0) return cmp;
        return Long.compare(this.id, o.id);
    }

    @Override public String toString() {
        return String.format("Event(t=%.6f, type=%s, server=%s, job=%d, class=%d)",
                time, type, server, jobId, jobClass);
    }
    public static void reset() {
        SEQ.set(0);
    }
}